docs/blueprints/zombie-segments-cleaner.md

```diff
@@ -0,0 +1,98 @@
+
+The second command for deleting bad segments should use the result of first command. The command should allow specifying flags like DB connection string, file with results from the first command, and dry run mode. Each reported segment should be verified if it's not changed since detection report was done. If segment was not changed it should be deleted from pointerDB. Verification and deletion should be done as atomic operation to avoid race conditions. Command should print the report at the end: number of deleted segments, number of segments skipped because of being newer than reported, skipped segments paths. Execution with dry run flag should only print results but without deleting segments from the database. This command should be executed against production database.
+
+## Implementation
+
+Code should be placed in package `cmd/segment-reaper`. The first command for detecting bad segments should be named `detect`. The second command for deleting bad segments should be named `delete`.
+
+Implementation can incorporate `metainfo.PointerDB` and `Iterate` method to go over all segments in DB. 
+
+Proposal for keeping segments structures:
+```
+// key will represent projectID/bucketName/encryptedPath
+map[string]Object
+
+type Object struct {
+    // big.Int represents a bitmask of unlimited size
+    segments big.Int
+    // if skip is true segments from object should be removed from memory when last segment is found 
+    // or iteration is finished,
+    // mark it as true if one of the segments from this object is newer then specified threshold
+    skip     bool 
+}
+```
+
+Output of `segment-reaper detect` command should be CSV with list of segments. Each row will contain segment path and creation/modification date, and serialized pointer for CompareAndSwap function:
+```
+    projectID/segmentIndex/bucketName/encrypted_path;creation_date
+    projectID/segmentIndex/bucketName/encrypted_path1;creation_date
+    projectID/segmentIndex/bucketName/encrypted_path3;creation_date
+```
+**NOTE**: Separator `;` is a valid symbol in paths. When splitting these lines into path and date, we should be careful to make the split on the last `;` in the string instead of on the first one. Alternatively separator symbol can be changed to symbol that is not occuring in segment path.
+
+Two major steps for deleting segment is verification if the segment is newer than detected earlier and delete segment in one atomic operation. For deleting segment we should use `CompareAndSwap` method. Sample code:
+```
+    // read the pointer
+    pointerBytes, err := s.DB.Get(ctx, []byte(path))
```

Comment:
```
pointerBytes, pointer, err := s.GetWithBytes(ctx, []byte(path))
```

Reply:
done, thanks

Reply:
You also don't need the umarshaling after these changes

Reply:
we need a pointer to compare dates

Reply:
`GetWithBytes` already returns the pointer

Reply:
ah, done